### PR TITLE
Drt 150 show less info in mail report

### PR DESCRIPTION
**Pull request by Sebyd**

### CODE REVIEW

File: Equilobe.DailyReport.Models/ReportFrame/JiraAuthor.cs [hunk 1]
```diff
@@ -62,9 +63,12 @@ public string RemainingChartPixelWidthString
         public List<IssueDetailed> OpenTasks { get; set; }
         public List<IssueDetailed> InProgressTasksParents { get; set; }
         public List<IssueDetailed> OpenTasksParents { get; set; }
+        public List<IssueDetailed> UncompletedTasks { get; set; }
```
Comment (DanBadea) [inline Equilobe.DailyReport.Models/ReportFrame/JiraAuthor.cs]: RemainingTasks

File: Equilobe.DailyReport.Models/ReportFrame/JiraAuthor.cs [hunk 2]
```diff
@@ -62,9 +63,12 @@ public string RemainingChartPixelWidthString
...
 
         public int InProgressTasksCount { get; set; }
         public int OpenTasksCount { get; set; }
+        public int UncompletedTasksCount { get; set; }
```
Comment (DanBadea) [inline Equilobe.DailyReport.Models/ReportFrame/JiraAuthor.cs]: RemainingTaskCount

File: Equilobe.DailyReport.Models/ReportFrame/JiraAuthor.cs [hunk 3]
```diff
@@ -78,9 +82,16 @@ public string RemainingChartPixelWidthString
         public bool HasAssignedIssues { get; set; }
         public bool HasDayLogs { get; set; }
         public int MaxHourValue { get; set; }
+        public bool HasSprint { get; set; }
 
         public SummaryGuidelineInfo GuidelineInfo { get; set; }
 
+        public UncompletedTasks InProgressTasksModel { get; set; }
+        public UncompletedTasks OpenTasksModel { get; set; }
+        public UncompletedTasks UncompletedTasksModel { get; set; }
```
Comment (DanBadea) [inline Equilobe.DailyReport.Models/ReportFrame/JiraAuthor.cs]: Rename without Model and keep only these properties and remove the separate collections of IssueDetailed.

File: Equilobe.DailyReport.Models/ReportFrame/Timing.cs [hunk 4]
```diff
@@ -34,6 +34,8 @@ public double TotalTimeHours
         public string InProgressTasksTimeLeftString { get; set; }
         public int OpenTasksTimeLeftSeconds { get; set; }
         public string OpenTasksTimeLeftString { get; set; }
+        public int UncompletedTasksTimeLeftSeconds { get; set; }
```
Comment (DanBadea) [inline Equilobe.DailyReport.Models/ReportFrame/Timing.cs]: RemainingTasks ...

File: Equilobe.DailyReport.Models/Views/UncompletedTasks.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+﻿using Equilobe.DailyReport.Models.ReportFrame;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Equilobe.DailyReport.Models.Views
+{
+    public class UncompletedTasks
```
Comment (DanBadea) [inline Equilobe.DailyReport.Models/Views/UncompletedTasks.cs]: AuthorTasks

File: JiraReporter/Services/AuthorLoader.cs [hunk 6]
```diff
@@ -249,6 +278,62 @@ private void SetAuthorInProgressTasks()
...
+            var tasksCount = 0;
+            _currentAuthor.UncompletedTasks = new List<IssueDetailed>();
+
+            AddToUncompletedTasksList(ref tasksCount, _currentAuthor.InProgressTasksParents);
+            AddToUncompletedTasksList(ref tasksCount, _currentAuthor.OpenTasksParents);
+
+            SetUncompletedTasksAdditionalCount();
+        }
+
+        private void AddToUncompletedTasksList(ref int count, List<IssueDetailed> tasks)
```
Comment (DanBadea) [inline JiraReporter/Services/AuthorLoader.cs]: use _currentAuthor.UncompletedTasks.Count

File: JiraReporter/Services/AuthorLoader.cs [hunk 7]
```diff
@@ -249,6 +278,62 @@ private void SetAuthorInProgressTasks()
             _currentAuthor.InProgressTasksParents = _currentAuthor.InProgressTasksParents.OrderBy(priority => priority.Priority.id).ToList();
         }
 
+        private void SetAuthorUncompletedTasksSection()
+        {
+            _currentAuthor.Timing.UncompletedTasksTimeLeftSeconds = _currentAuthor.Timing.InProgressTasksTimeLeftSeconds + _currentAuthor.Timing.OpenTasksTimeLeftSeconds;
+            _currentAuthor.Timing.UncompletedTasksTimeLeftString = _currentAuthor.Timing.UncompletedTasksTimeLeftSeconds.SetTimeFormat8Hour();
+            var tasksCount = 0;
```
Comment (DanBadea) [inline JiraReporter/Services/AuthorLoader.cs]: Make it a class field and remove the ref usage.

File: JiraReporter/Services/AuthorLoader.cs (hunk 6 above)
Comment (DanBadea) [inline JiraReporter/Services/AuthorLoader.cs]: refactor this method. make it simpler.

File: JiraReporter/Services/AuthorLoader.cs [hunk 8]
```diff
@@ -249,6 +278,62 @@ private void SetAuthorInProgressTasks()
...
+                    }
+                }
+
+                _currentAuthor.UncompletedTasks.Add(newIssue);
+            }
+        }
+
+        private void SetUncompletedTasksAdditionalCount()
+        {
+            var uncompletedVisibleTasksCount = _currentAuthor.UncompletedTasks.Count + _currentAuthor.UncompletedTasks.SelectMany(t => t.SubtasksDetailed).ToList().Count;
```
Comment (DanBadea) [inline JiraReporter/Services/AuthorLoader.cs]: Some of the UncompletedTasks may be assigned to someone else.

File: JiraReporter/Services/AuthorLoader.cs [hunk 9]
```diff
@@ -274,7 +359,7 @@ private void SetAuthorDayLogs()
             foreach (var day in _options.ReportDates)
                 _currentAuthor.DayLogs.Add(DayLogLoader.CreateDayLog(_currentAuthor, day, _context));
             _currentAuthor.DayLogs = _currentAuthor.DayLogs.OrderBy(d => d.Date).ToList();
-            _currentAuthor.DayLogs.RemoveAll(d => d.Commits.Count == 0 && d.Issues == null);
+            _currentAuthor.DayLogs.RemoveAll(d => d.Commits.Count == 0 && (d.Issues == null || d.Issues.Count == 0));
```
Comment (DanBadea) [inline JiraReporter/Services/AuthorLoader.cs]: You can create a helper method for collections `IsEmpty` to check for both null and `count = 0`.

File: JiraReporter/Views/IndividualReportTemplate.cshtml [hunk 10]
```diff
@@ -44,7 +44,7 @@ else
                                     <tbody>
                                         <tr>
                                             <td style="vertical-align:top; width:15px">
-                                                <span>&#x25cf;</span>
+                                                <span><span style="font-style:normal; font-size:25px; vertical-align:middle">&#x2022; </span></span>
```
Comment (DanBadea) [inline JiraReporter/Views/IndividualReportTemplate.cshtml]: Create partial view for the dot. 

You will need an override for the `Partial` method on the `ExtendedTemplateBase`.

File: JiraReporter/Views/TimesheetReportTemplate.cshtml [hunk 11]
```diff
@@ -1,225 +1,62 @@
...
-                            or
-                        </td>
-                        <td width="100px" height="40px;" style="display:inline-block; padding:0px; line-height:40px;">
-                            <a style="color:#0000FF; font-weight:bold; text-decoration:none; font-size:12px; line-height:40px" href="@Model.SendDraftUrl" title="If the information in the report has changed and you want to see how it looks now, before confirming it, you can get a new one.">Resend</a>
-                        </td>
-                    </tr>
-                </tbody>
-            </table>
-    }
+                @if (Model.IsFinalDraft)
```
Comment (DanBadea) [inline JiraReporter/Views/TimesheetReportTemplate.cshtml]: Extract as partial view

File: JiraReporter/Views/footer.cshtml [hunk 12]
```diff
@@ -0,0 +1,22 @@
+﻿<table style="margin-top:50px">
+    <tbody>
+        <tr>
+            <td height="25px" style="max-height:25px"> <img style="width:100px; height:25px" src="https://dailyreport.equilobe.com/content/images/logo-menu.PNG" /> </td>
+            <td height="25px" style="max-height:25px"></td>
+            <td height="25px" style="max-height:25px"></td>
+            <td height="25px" style="max-height:25px"></td>
```
Comment (DanBadea) [inline JiraReporter/Views/footer.cshtml]: One td per tr is OK for the footer.

File: Equilobe.DailyReport.Models/Views/UncompletedTasks.cs (hunk 5 above)
Comment (Sebyd) [inline Equilobe.DailyReport.Models/Views/UncompletedTasks.cs]: Se merita sa extindem List? 
Din cate am vazut trebuie rescrise metodele daca vrem sa le folosim. Gen 'ToList()'

Comment (DanBadea) [inline Equilobe.DailyReport.Models/Views/UncompletedTasks.cs]: Vezi ca noi nu extindem List ci List<IssueDetailed> si sunt aprope sigur ca nu trebuie sa faci nimic in plus.

Comment (DanBadea) [inline Equilobe.DailyReport.Models/Views/UncompletedTasks.cs]: Vezi ca noi nu extindem List ci List<IssueDetailed> si sunt aprope sigur ca nu trebuie sa faci nimic in plus.

Comment (DanBadea) [inline Equilobe.DailyReport.Models/Views/UncompletedTasks.cs]: Vezi ca noi nu extindem `List` ci `List<IssueDetailed>` si sunt aprope sigur ca nu trebuie sa faci nimic in plus.

**Pull request merged**